dev/import-beats/kibana.go

```diff
@@ -0,0 +1,220 @@
+		}
+	}
+	return ms, nil
+}
+
+func createKibanaContent(kibanaMigrator *kibanaMigrator, modulePath string) (kibanaContent, error) {
+	moduleDashboardPath := path.Join(modulePath, "_meta", "kibana", "7", "dashboard")
+	moduleDashboards, err := ioutil.ReadDir(moduleDashboardPath)
+	if os.IsNotExist(err) {
+		log.Printf("\tno dashboards present, skipped (modulePath: %s)", modulePath)
+		return kibanaContent{}, nil
+	} else if err != nil {
+		return kibanaContent{}, errors.Wrapf(err, "reading module dashboard directory failed (path: %s)",
+			moduleDashboardPath)
+	}
+
+	kibana := kibanaContent{
+		dashboardFiles:     map[string][]byte{},
+		visualizationFiles: map[string][]byte{},
+	}
+	for _, moduleDashboard := range moduleDashboards {
+		log.Printf("\tdashboard found: %s", moduleDashboard.Name())
+
+		dashboardFilePath := path.Join(moduleDashboardPath, moduleDashboard.Name())
+		dashboardFile, err := ioutil.ReadFile(dashboardFilePath)
+		if err != nil {
+			return kibanaContent{}, errors.Wrapf(err, "reading dashboard file failed (path: %s)",
+				dashboardFilePath)
+		}
+
+		migrated, err := kibanaMigrator.migrateDashboardFile(dashboardFile)
+		if err != nil {
+			return kibanaContent{}, errors.Wrapf(err, "migrating dashboard file failed (path: %s)",
+				dashboardFilePath)
+		}
+
+		extractedDashboards, err := extractKibanaObjects(migrated, "dashboard")
+		if err != nil {
+			return kibanaContent{}, errors.Wrapf(err, "extracting kibana dashboards failed")
+		}
+
+		for k, v := range extractedDashboards {
+			kibana.dashboardFiles[k] = v
+		}
+
+		extractedVisualizations, err := extractKibanaObjects(migrated, "visualization")
+		if err != nil {
+			return kibanaContent{}, errors.Wrapf(err, "extracting kibana visualizations failed")
+		}
+
+		for k, v := range extractedVisualizations {
+			kibana.visualizationFiles[k] = v
+		}
+	}
+	return kibana, nil
+}
+
+func extractKibanaObjects(dashboardFile []byte, objectType string) (map[string][]byte, error) {
+	var documents kibanaDocuments
+
+	err := json.Unmarshal(dashboardFile, &documents)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling migrated dashboard file failed")
+	}
+
+	extracted := map[string][]byte{}
+	for _, object := range documents.Objects {
+		aType, err := object.getValue("type")
+		if err != nil {
+			return nil, errors.Wrapf(err, "retrieving type failed")
+		}
+
+		if aType != objectType {
+			continue
+		}
+
+		err = object.delete("updated_at")
+		if err != nil {
+			return nil, errors.Wrapf(err, "removing field updated_at failed")
+		}
+
+		err = object.delete("version")
+		if err != nil {
+			return nil, errors.Wrapf(err, "removing field version failed")
+		}
+
+		data, err := json.MarshalIndent(object, "", "    ")
+		if err != nil {
+			return nil, errors.Wrapf(err, "marshalling object failed")
+		}
+
+		id, err := object.getValue("id")
+		if err != nil {
+			return nil, errors.Wrapf(err, "retrieving id failed")
+		}
+
+		extracted[id.(string)+".json"] = data
```

Comment:
The id of the saved object becomes the name? I like that.

Reply:
That is true :)